observability/src/test/java/io/grpc/observability/interceptors/InternalLoggingChannelInterceptorTest.java

```diff
@@ -156,122 +169,123 @@ public String authority() {
+          eq("the-authority"),
+          ArgumentMatchers.<Duration>isNull(),
+          same(clientInitial),
+          eq(filterParams.headerBytes),
+          eq(EventLogger.LOGGER_CLIENT),
+          anyString(),
+          ArgumentMatchers.<SocketAddress>isNull());
+      verifyNoMoreInteractions(mockLogHelper);
       assertSame(clientInitial, actualClientInitial.get());
     }
 
-    // TODO(dnvindhya) : Add a helper method to verify other fields of GrpcLogRecord for all events
     // receive server header
     {
-      EventType expectedResponseHeaderEvent = EventType.GRPC_CALL_RESPONSE_HEADER;
-      ArgumentCaptor<GrpcLogRecord> captor = ArgumentCaptor.forClass(GrpcLogRecord.class);
       Metadata serverInitial = new Metadata();
       interceptedListener.get().onHeaders(serverInitial);
-      verify(mockSink, times(2)).write(captor.capture());
-      assertEquals(captor.getValue().getEventType(),
-          expectedResponseHeaderEvent);
-      verifyNoMoreInteractions(mockSink);
+      verify(mockLogHelper).logResponseHeader(
+          /*seq=*/ eq(2L),
+          eq("service"),
+          eq("method"),
+          same(serverInitial),
+          eq(filterParams.headerBytes),
+          eq(EventLogger.LOGGER_CLIENT),
+          anyString(),
+          same(peer));
+      verifyNoMoreInteractions(mockLogHelper);
       verify(mockListener).onHeaders(same(serverInitial));
     }
 
     // send client message
     {
-      EventType expectedRequestMessageEvent = EventType.GRPC_CALL_REQUEST_MESSAGE;
-      ArgumentCaptor<GrpcLogRecord> captor = ArgumentCaptor.forClass(GrpcLogRecord.class);
       byte[] request = "this is a request".getBytes(US_ASCII);
       interceptedLoggingCall.sendMessage(request);
-      verify(mockSink, times(3)).write(captor.capture());
-      assertEquals(captor.getValue().getEventType(),
-          expectedRequestMessageEvent);
-      verifyNoMoreInteractions(mockSink);
+      verify(mockLogHelper).logRpcMessage(
+          /*seq=*/ eq(3L),
+          eq("service"),
+          eq("method"),
+          eq(EventType.GRPC_CALL_REQUEST_MESSAGE),
+          same(request),
+          eq(filterParams.messageBytes),
+          eq(EventLogger.LOGGER_CLIENT),
+          anyString());
+      verifyNoMoreInteractions(mockLogHelper);
       assertSame(request, actualRequest.get());
     }
 
     // client half close
     {
-      EventType expectedHalfCloseEvent = EventType.GRPC_CALL_HALF_CLOSE;
-      ArgumentCaptor<GrpcLogRecord> captor = ArgumentCaptor.forClass(GrpcLogRecord.class);
       interceptedLoggingCall.halfClose();
-      verify(mockSink, times(4)).write(captor.capture());
-      assertEquals(captor.getValue().getEventType(),
-          expectedHalfCloseEvent);
+      verify(mockLogHelper).logHalfClose(
+          /*seq=*/ eq(4L),
+          eq("service"),
+          eq("method"),
+          eq(EventLogger.LOGGER_CLIENT),
+          anyString());
       halfCloseCalled.get(1, TimeUnit.SECONDS);
```

Comment:
DO we need a timeout of 1 second? As far as I can see there is no async operation happening on `halfCloseCalled`.

Reply:
As per our discussion offline, reduced the Timeout and keeping it as `SettableFuture` for now.